index.js

```diff
@@ -21,7 +21,7 @@ function querystring(query) {
   //
   for (;
     part = parser.exec(query);
-    result[decodeURIComponent(part[1])] = decodeURIComponent(part[2])
+    result[decodeURIComponent(part[1])] = part[2] ? decodeURIComponent(part[2]) : null
```

Comment:
I wonder if it's better to keep it as is. When there isn't a value an empty string will be used instead of `null`.
This is consistent with the Node.js querystring module and other popular modules like `qs`:
```js
> querystring.parse('foo&bar=baz')
{ foo: '', bar: 'baz' }
> querystring.parse('foo=&bar=baz')
{ foo: '', bar: 'baz' }
```